iOS/LocalPackages/DataBrokerProtection-iOS/Sources/DataBrokerProtection-iOS/ManagingAndCoordinating/DataBrokerProtectionIOSManager.swift

```diff
@@ -348,6 +343,25 @@ extension DataBrokerProtectionIOSManager: JobQueueManagerDelegate {
             }
         }
     }
+
+    public func queueManagerDidCompleteIndividualJob(_ queueManager: any DataBrokerProtectionCore.JobQueueManaging) {
+
+        // Figure out if we've just finished initial scans, and send the appropriate pixel if necessary
+        if eventPixels.hasInitialScansTotalDurationPixelBeenSent() {
+            return
+        }
+
+        do {
+            let hasCompletedInitialScans = try database.haveAllScansRunAtLeastOnce()
```

Comment:
At the risk of premature optimization, isn’t it a costly operation? We have to fetch all broker profile query data and use Swift to loop through them. This happens for every operation during initial scans which adds up. Maybe using a raw SQL query if it’s just to check lastRunDate?

Reply:
Yeah it's fairly costly. I figure it's okay given a single operation is on the order of many seconds. For me the bigger reason to change it is to stop using the BrokerProfileQueryData object for everything, although in this case it's okay since it's confined to the DB method. 

I figure we can change it when we're in slightly less expedient times